store/src/java/com/zimbra/util/ExternalVolumeInfoHandler.java

```diff
@@ -102,15 +113,19 @@ public void deleteServerProperties(int volumeId, String serverId) throws Service
             // step 6: Set ldap attribute
             provisioning.getLocalServer().setServerExternalStoreConfig(currentJsonObject.toString());
 
-            editGlobalConfigOnDeleteVolume(volumeId, serverId);
-
+            //if volume is unified, then only we should call the below method to modify global external store
+            if (isUnifiedVolume) {
+                editGlobalConfigOnDeleteVolume(volumeId, serverId);
```

Comment:
can't we call flushConfigLevelCacheOnAllServers from here itself? we are changing function signatures to return isUnifiedVolume flag and at the end based on that flag calling flushConfigLevelCacheOnAllServers, which can be done here also

Reply:
Yes right, we could have done it by 2 ways: 
1. Carry the AdminDocumentHandler reference as a parameter from DeleteVolume API up to here to handler class. and
2. to send back the boolean value from handler.
I opted for option 2, to maintain the uniformity with other API, as we are calling `flushConfigLevelCacheOnAllServers` method from API's itself.

Please suggest if I should change it.?

Reply:
ok I missed the handler part, option 1 is ok